### DISCUSSION TITLE
Filter bills by actions

### DISCUSSION THREAD

**Discussion by dannyfritz** [openstates/issues#26] [Proposals & RFCs]
**Is your feature request related to a problem? Please describe.**
I want to make a website that displays all bills in Colorado that have passed both the senate and house, but haven't been signed by the governor yet.

**Describe the solution you'd like**
I'd like to be able to filter bills by which actions it has passed/failed.
* Filter by "passed senate committee"
* Filter by "passed house committee and passed senate committee"
* Filter by "failed house committee"
* etc.

**Describe alternatives you've considered**
Fetch all bill data, page it all, and filter it on the consumer side.
Download the bulk data and filter it on the consumer side.

**Additional context**
https://discourse.openstates.org/t/api-v2-filter-by-status/490

**Comment by jamesturk** [comment]
This is likely possible to do using the raw data now based on action categorization, but the completeness of action categorization varies a lot per state.

There's also the issue that action categories are independent from chambers, so to query actions generically it'd look something like:
`
(has_action=passed, in_chamber=upper) & (has_action=passed, in_chamber=lower)
`
which right now is richer than the query language supports

There's the simpler option, of just allowing certain named combinations:

has_passed_upper, has_passed_lower, has_been_signed, etc.

This might be simpler, but less flexible.

Committees, mentioned in the parent, are even harder, since a bill can pass multiple committees, there's not a clear way to define "passed_lower_committee" correctly if it has to travel through more than one.

**Comment by jamesturk** [comment]
I'm curious if any of those interested in this ticket have opinions on the tradeoff vs. a generic way to query actions vs. some specific subset.  I'd also be more than glad to guide someone through working on this feature if there's interest.
